exchanges/stream/websocket_connection.go

```diff
@@ -12,17 +12,32 @@ import (
 	"net"
 	"net/http"
 	"net/url"
+	"sync"
 	"sync/atomic"
 	"time"
 
 	"github.com/gorilla/websocket"
 	"github.com/thrasher-corp/gocryptotrader/log"
 )
 
+// WebsocketConnection contains all the data needed to send a message to a WS
+// connection
+type WebsocketConnection struct {
+	// Gorilla websocket does not allow more than one goroutine to utilise writes methods
+	writeControl     sync.Mutex
+	connected        int32
+	connection       *websocket.Conn
+	rateLimit        int64
+	_URL             string
+	sharedContext    *Websocket
```

Comment:
Sorry I don't like `sharedContext`, doesn't seem fitting. `parentWebsocket` or `parent`

Reply:
Yeah good suggestion will change, but why you give me 2 choices, because I will always pick the less length one. 😆